src/entry_table.rs

```diff
@@ -227,12 +231,16 @@ impl<const UID_LENGTH: usize, const KWI_LENGTH: usize> EntryTableValue<UID_LENGT
         max_results: usize,
         kwi_chain_table_uids: &mut KwiChainUids<UID_LENGTH, KWI_LENGTH>,
     ) {
-        let kwi_uid: KmacKey = self.kwi.derive_kmac_key(CHAIN_TABLE_KEY_DERIVATION_INFO);
-
         let entry = kwi_chain_table_uids
             .entry(self.kwi.clone())
             .or_insert_with(Vec::new);
 
+        if self.chain_table_uid.is_none() {
+            return;
+        }
+
+        let kwi_uid: KmacKey = self.kwi.derive_kmac_key(CHAIN_TABLE_KEY_DERIVATION_INFO);
+
         // derive the Chain Table UID
```

Comment:
Maybe you can suggest change for this kind of remark:
```suggestion
        // Derive the Chain Table UID.
```
Otherwise I may be too lazy to correct all comments one by one. Especially if this is not new code.

Reply:
Great feature, will do.